sympy/tensor/array/arrayop.py

```diff
@@ -130,9 +130,16 @@ def tensorcontraction(array, *contraction_axes):
     for icontrib in itertools.product(*remaining_indices):
         i = sum(icontrib)
         isum = S.Zero
+        summed_deltas = []
         for axes_group in contraction_axes:
```

Comment:
This loop, computing ``summed_deltas``,  appears independent of ``i = sum(icontrib)``. Could it be taken out of the ``icontrib`` loop?